Sources/NIO/SocketAddresses.swift

```diff
@@ -325,6 +327,34 @@ public enum SocketAddress: CustomStringConvertible {
             throw SocketAddressError.failedToParseIPString(ipAddress)
         }
     }
+    
+    /// Create a new `SocketAddress` for an IP address in ByteBuffer form.
+    ///
+    /// - parameters:
+    ///     - ipAddress: The IP address, in ByteBuffer form.
+    ///     - port: The target port.
+    /// - returns: the `SocketAddress` corresponding to this string and port combination.
+    /// - throws: may throw `SocketAddressError.failedToParseIPByteBuffer` if the IP address cannot be parsed or `SocketAddressError.unsupported` if the ATYP is wrong.
```

Comment:
We aren't using an `ATYP` here and can never throw `unsupported`. Can we update this throws message?